packages/angular/build/src/utils/load-proxy-config.ts

```diff
@@ -49,33 +47,8 @@ export async function loadProxyConfiguration(
 
       break;
     }
-    case '.mjs':
-      // Load the ESM configuration file using the TypeScript dynamic import workaround.
-      // Once TypeScript provides support for keeping the dynamic import this workaround can be
-      // changed to a direct dynamic import.
-      proxyConfiguration = await loadEsmModule<{ default: unknown }>(pathToFileURL(proxyPath));
-      break;
-    case '.cjs':
-      proxyConfiguration = require(proxyPath);
-      break;
     default:
-      // The file could be either CommonJS or ESM.
-      // CommonJS is tried first then ESM if loading fails.
-      try {
-        proxyConfiguration = require(proxyPath);
-        break;
-      } catch (e) {
-        assertIsError(e);
-        if (e.code === 'ERR_REQUIRE_ESM' || e.code === 'ERR_REQUIRE_ASYNC_MODULE') {
```

Comment:
We should still handle the async case here (top-level await in the config file or its dependencies). Unfortunately, we would still need the workaround for this one case.